### PR TITLE
Improve compact readability

### PR DESCRIPTION
**Pull request by AllenzhLi**
* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

### CODE REVIEW

File: pkg/compact/downsample/downsample.go [hunk 1]
```diff
@@ -117,8 +117,8 @@ func Downsample(
 			return id, errors.Wrapf(err, "get series %d", postings.At())
 		}
 
-		for i, c := range chks[1:] {
-			if chks[i].MaxTime >= c.MinTime {
+		for _, c := range chks[1:] {
+			if c.MaxTime >= c.MinTime {
```
Comment (yeya24) [inline pkg/compact/downsample/downsample.go:121]: I think this shouldn't be changed. The logic is different

**Review by yeya24 — CHANGES_REQUESTED** [review]

File: pkg/compact/compact.go [hunk 2]
```diff
@@ -380,8 +380,8 @@ func (cg *Group) Key() string {
 	return cg.key
 }
 
-// Add the block with the given meta to the group.
-func (cg *Group) Add(meta *metadata.Meta) error {
+// AppendMeta the block with the given meta to the group.
```
Comment (yeya24) [inline pkg/compact/compact.go:383]: The method name is changed, please update the comment, too

File: pkg/compact/downsample/downsample.go (hunk 1 above)
Comment (AllenzhLi) [reply pkg/compact/downsample/downsample.go]: Can you explain it in detail?

File: pkg/compact/compact.go (hunk 2 above)
Comment (AllenzhLi) [reply pkg/compact/compact.go:383]: `// AppendMeta the block with the given meta to the group.
func (cg *Group) AppendMeta(meta *metadata.Meta) error {
	cg.mtx.Lock()
	defer cg.mtx.Unlock()

	if !labels.Equal(cg.labels, labels.FromMap(meta.Thanos.Labels)) {
		return errors.New("block and group labels do not match")
	}
	if cg.resolution != meta.Thanos.Downsample.Resolution {
		return errors.New("block and group resolution do not match")
	}

	cg.metasByMinTime = append(cg.metasByMinTime, meta)
	sort.Slice(cg.metasByMinTime, func(i, j int) bool {
		return cg.metasByMinTime[i].MinTime < cg.metasByMinTime[j].MinTime
	})
	return nil
}`

The comment is aleady update and same with func name.

File: pkg/compact/downsample/downsample.go (hunk 1 above)
Comment (yeya24) [reply pkg/compact/downsample/downsample.go]:
``` go
		for i, c := range chks[1:] {
			if chks[i].MaxTime >= c.MinTime {
```

Here it is comparing the chks[i].MaxTime and chks[i + 1].MinTime. Not the same chunk.

Comment (AllenzhLi) [reply pkg/compact/downsample/downsample.go]: Got it

File: pkg/compact/compact.go (hunk 2 above)
Comment (yeya24) [reply pkg/compact/compact.go]: Yes, but I mean the sentence has a grammar issue. Please add a verb to make it readable.

Comment (AllenzhLi) [reply pkg/compact/compact.go]: Understood, has been updated

File: pkg/compact/compact.go [hunk 3]
```diff
@@ -380,8 +380,8 @@ func (cg *Group) Key() string {
 	return cg.key
 }
 
-// Add the block with the given meta to the group.
-func (cg *Group) Add(meta *metadata.Meta) error {
+// AppendMeta add the block with the given meta to the compaction group.
```
Comment (yeya24) [inline pkg/compact/compact.go:383]: `AppendMeta adds`

File: pkg/block/fetcher.go [hunk 4]
```diff
@@ -530,7 +530,7 @@ func NewLabelShardedMetaFilter(relabelConfig []*relabel.Config) *LabelShardedMet
 	return &LabelShardedMetaFilter{relabelConfig: relabelConfig}
 }
 
-// Special label that will have an ULID of the meta.json being referenced to.
+// BlockIDLabel Special label that will have an ULID of the meta.json being referenced to.
 const BlockIDLabel = "__block_id"
```
Comment (yeya24) [inline pkg/block/fetcher.go:534]: This change is unnecessary. The original comment is better.

File: pkg/compact/compact.go (hunk 3 above)
Comment (AllenzhLi) [reply pkg/compact/compact.go]: Got it.

File: pkg/block/fetcher.go (hunk 4 above)
Comment (AllenzhLi) [reply pkg/block/fetcher.go]: Restored.

File: cmd/thanos/compact.go [hunk 5]
```diff
@@ -343,16 +371,20 @@ func runCompact(
...
 		defer cleanMtx.Unlock()
 
-		if err := sy.SyncMetas(ctx); err != nil {
+		if err := syncer.SyncMetas(ctx); err != nil {
 			cancel()
 			return errors.Wrap(err, "syncing metas")
 		}
 
-		compact.BestEffortCleanAbortedPartialUploads(ctx, logger, sy.Partial(), bkt, partialUploadDeleteAttempts, blocksCleaned, blockCleanupFailures)
+		compact.BestEffortCleanAbortedPartialUploads(ctx, logger, syncer.Partial(), bkt,
```
Comment (bwplotka) [inline cmd/thanos/compact.go:379]: ditto

File: cmd/thanos/compact.go [hunk 6]
```diff
@@ -297,20 +323,22 @@ func runCompact(
 		conf.acceptMalformedIndex,
 		enableVerticalCompaction,
 		reg,
-		blocksMarked.WithLabelValues(metadata.DeletionMarkFilename),
-		garbageCollectedBlocks,
+		compactMetrics.blocksMarked.WithLabelValues(metadata.DeletionMarkFilename),
+		compactMetrics.garbageCollectedBlocks,
+	)
+	blocksCleaner := compact.NewBlocksCleaner(logger, bkt, ignoreDeletionMarkFilter, deleteDelay,
```
Comment (bwplotka) [inline cmd/thanos/compact.go:329]: ditto

File: cmd/thanos/compact.go [hunk 7]
```diff
@@ -90,63 +90,82 @@ func registerCompact(app *extkingpin.App) {
...
+
+func newCompactMetrics(reg *prometheus.Registry, deleteDelay time.Duration) *CompactMetrics {
+	_ = promauto.With(reg).NewGaugeFunc(prometheus.GaugeOpts{
+		Name: "thanos_delete_delay_seconds",
+		Help: "Configured delete delay in seconds.",
+	}, func() float64 {
+		return deleteDelay.Seconds()
+	})
+
+	m := new(CompactMetrics)
```
Comment (bwplotka) [inline cmd/thanos/compact.go:113]: Let's stick to `m := CompactMetrics{}` for consistency (we never use new). I think we don't need pointer too, that's why new is not needed

**Review by bwplotka — APPROVED** [review]
It looks amazing, thanks! I have few comments, but otherwise LGTM, thanks for keeping logic the same since it's refactoring 🤗

**Comment by bwplotka** [comment]
Friendly ping (:

File: cmd/thanos/compact.go [hunk 8]
```diff
@@ -90,63 +90,82 @@ func registerCompact(app *extkingpin.App) {
...
-	logger log.Logger,
-	tracer opentracing.Tracer,
-	reg *prometheus.Registry,
-	component component.Component,
-	conf compactConfig,
-	flagsMap map[string]string,
-) error {
-	deleteDelay := time.Duration(conf.deleteDelay)
-	halted := promauto.With(reg).NewGauge(prometheus.GaugeOpts{
+type CompactMetrics struct {
```
[the remark this answers is not in the document]
Comment (AllenzhLi) [reply cmd/thanos/compact.go:93]: no，I will update it to compactMetrics

**Comment by AllenzhLi** [comment]
Unfortunately, I deleted the original fork.

I have resubmit the pr in 4016 and I will close this pr.

**Pull request closed without merging**